app/utils.py

```diff
@@ -1,28 +1,47 @@
 import logging
 import os
-import yaml
+import json
 from typing import Union
 
 
 logger = logging.getLogger('app-logger')
 
 
-def generate_email_map() -> Union[dict, None]:
+def generate_email_map() -> "Union[dict[str, int], None]":
     """
-    returns a map of lowercased user emails
-    to the corresponding file path where their 
-    user config yml file is
+    Returns a map of lowercased user emails
+    to the corresponding user-config index in `DIRECTORY_FILE_PATH`. 
+    Note that the file at `DIRECTORY_FILE_PATH` is a
+    list of dicts (list of user configs).
     """
-    path = os.getenv("DIRECTORY_PATH")
+    path = os.getenv("DIRECTORY_FILE_PATH")
     if path is None:
         logger.error("Env var to user configs not set!")
         return
     logger.info(f"Path to user configs: {path}")
     email_map = {}
-    for file in os.scandir(path):
-        user_config = yaml.safe_load(open(file))
-        contact_emails: list[str] = user_config["general"]["contact_emails"]
-        for contact_email in contact_emails:
-            contact_email = contact_email.lower()
-            email_map[contact_email] = os.path.abspath(file)
+    with open(path) as f:
+        user_directory: list[dict] = json.load(f)
+
+    for idx, user in enumerate(user_directory):
+        try:
+            contact_emails: list[str] = user["general"]["contact_emails"]
+            for contact_email in contact_emails:
+                contact_email = contact_email.lower()
+                email_map[contact_email] = idx
```

Comment:
Whoa interesting that we are using indices to do stuff. I don't think it's too much overhead to just store a reference to the entire user here, because that would just store a pointer to the data anyway. Even if we duplicate the data it's a small piece of data 😄

Reply:
Okies, sounds good